lib/Controller/OpenAiAPIController.php

```diff
@@ -147,14 +164,65 @@ public function getImageGenerationContent(string $hash, int $urlId): DataDisplay
 	/**
 	 * @param string $hash
 	 * @return TemplateResponse
-	 * @throws Exception
-	 * @throws MultipleObjectsReturnedException
 	 */
 	#[NoAdminRequired]
 	#[NoCSRFRequired]
 	public function getImageGenerationPage(string $hash): TemplateResponse {
-		$generationData = $this->openAiAPIService->getGenerationInfo($hash);
+		try {
+			$generationData = $this->openAiAPIService->getGenerationInfo($hash);
+		} catch (Exception $e) {
+			$generationData['error'] = $e->getMessage();
+		}
 		$this->initialStateService->provideInitialState('generation', $generationData);
 		return new TemplateResponse(Application::APP_ID, 'imageGenerationPage');
 	}
+
+	/**
+	 * @param bool|null $clearTextPrompts
+	 * @param bool|null $clearImagePrompts
+	 * @return DataResponse
+	 */
+	#[NoAdminRequired]
+	public function clearPromptHistory(?bool $clearTextPrompts = null, ?bool $clearImagePrompts = null): DataResponse {
+		$this->logger->debug('Clearing prompt history: ' . 'clearTextPrompts ' . strval($clearTextPrompts) . ' clearImagePrompts ' . strval($clearImagePrompts));
+		if ($clearTextPrompts === true) {
+			try {
+				$this->openAiAPIService->clearPromptHistory($this->userId, Application::PROMPT_TYPE_TEXT);
+			} catch (DBException $e) {
+				return new DataResponse($this->l10n->t('Unknown error while clearing prompt history'), Http::STATUS_BAD_REQUEST);
```

Comment:
I guess one could take the stance that it doesn't matter what error response code you use as long as it's an error code. Still, semantically the client did nothing wrong if the database fails, which would make it a 5xx status. I'm not saying you should make sure to always get the most specific status code you can find, but distinguishing 500 vs 400 would be nice IMHO :shrug:

Reply:
Sounds sensible, I'll change the error handling such that it returns 5xx for server side failures.

Reply:
Done: https://github.com/nextcloud/integration_openai/pull/54/commits/630648b4d240eefce15c27ef9ad6153acd5928f8